src/source/Common/RequestInfo.c

```diff
@@ -245,7 +245,8 @@ STATUS removeRequestHeader(PRequestInfo pRequestInfo, PCHAR headerName)
 
         if (STRCMPI(pCurrentHeader->pName, headerName) == 0) {
             CHK_STATUS(singleListDeleteNode(pRequestInfo->pRequestHeaders, pCurNode));
-
+            // Free only if found
+            SAFE_MEMFREE(pCurrentHeader);
             // Early return
             CHK(FALSE, retStatus);
         }
```

Comment:
Ok this logic is only functionally different in the case that we executed these lines and 

```
CHK_STATUS(singleListGetNodeData(pCurNode, &item));
pCurrentHeader = (PRequestHeader) item;
```

```
STRCMPI(pCurrentHeader->pName, headerName) == 0
```

Is `FALSE` in that case previously we would have called SAFE_MEMFREE, but now we do not.  Do we have a targeted test for this case? 

In other places we do a null check right after this line:
```
pCurrentHeader = (PRequestHeader) item;
```

Do we not need that were as well since we deref `pCurrentHeader` later or as we just assuming it can never be null (perhaps there is validation on every insert and we have mutexes to make sure it cannot be made null by another thread?). If the header is allocated but it didn't match the `STRCMPI` it doesn't get free I'm concerned about a leak in that case because I'm not seeing where that would get freed?

Reply:
I understand now thank you!